src/viking.js

```diff
@@ -1,11 +1,86 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health,strength){
+        this.health = health;
+        this.strength = strength;
+    }
 
+    attack() {
+
+        return this.strength
+    }
+
+    receiveDamage(damage){
+        this.health -= damage
+        
+    }
+
+}
 // Viking
-class Viking {}
+
+class Viking extends Soldier{
+    constructor(name,health,strength){
+        super(health,strength)
+        this.name = name;
+    }
+    receiveDamage(damage){
+        this.health -= damage
+        if(this.health > 0){
+            return `${this.name} has received ${damage} points of damage`
+        }else{
+            return `${this.name} has died in act of combat`
+        }
+    }
+    
+    battleCry(){
+        return  "Odin Owns You All!"
+    }
+
+}
 
 // Saxon
-class Saxon {}
+
+class Saxon extends Soldier{
+    constructor(health,strength){
+       super( health,strength) 
+    }
+    receiveDamage(damage){
+        this.health -= damage
+        if(this.health > 0){
+            return `A Saxon has received ${damage} points of damage`
+        }else{
+            return `A Saxon has died in combat`
+        }
+    }
+}
 
 // War
-class War {}
+class War {
+       vikingArmy = [];
+       saxonArmy = [];
+
+    //static get [Symbol.species]() { return Array; }
+    
+ 
+    addViking(Viking) {
+        this.vikingArmy = [Viking]
```

Comment:
Funciona pero estás creando un nuevo array con Viking dentro.
Para conservar el array con el que se ha inicializado this.vikingArmy (en el constructor), se usa el método push:
`this.vikingArmy.push(Viking)`